mbtools/html_content_extractor.py

```diff
@@ -0,0 +1,47 @@
+from mbtools import utils
+from pathlib import Path
+import argparse
+
+
+class ContentExtractor:
+    """
+    Given a string with path to an extracted moodle backup directory return
+    model objects for course html elements
+    """
+    def __init__(self, mbz_path):
+        self.mbz_path = Path(mbz_path)
+        self.output_html_files = {}
+
+    def replace_content_tags(self, output_file_path):
+        activities = utils.parse_backup_activities(self.mbz_path)
+
+        for act in activities:
+            for html_elem in act.html_elements():
+
+                self.output_html_files.update(html_elem.replace_content_tag())
+                with open(act.activity_filename, "wb") as f:
```

Comment:
I think this will write out every file, even if no content was extracted which may lead to noisier change sets than strictly necessary. Could we instead only write out activity files where content was modified?